### ISSUE TITLE
 AttributeError: module 'urllib' has no attribute 'quote'

### ISSUE THREAD

**Issue by angeloudy** [saltstack/salt#47534]
### Description of Issue/Question
```
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/local/lib/python3.6/site-packages/salt/minion.py", line 1493, in _thread_return
        return_data = executor.execute()
      File "/usr/local/lib/python3.6/site-packages/salt/executors/direct_call.py", line 28, in execute
        return self.func(*self.args, **self.kwargs)
      File "/usr/local/lib/python3.6/site-packages/salt/modules/state.py", line 603, in apply_
        return sls(mods, **kwargs)
      File "/usr/local/lib/python3.6/site-packages/salt/modules/state.py", line 1141, in sls
        ret = st_.state.call_high(high_, orchestration_jid)
      File "/usr/local/lib/python3.6/site-packages/salt/state.py", line 2539, in call_high
        ret = self.call_chunks(chunks)
      File "/usr/local/lib/python3.6/site-packages/salt/state.py", line 2030, in call_chunks
        if self.reconcile_procs(running):
      File "/usr/local/lib/python3.6/site-packages/salt/state.py", line 2058, in reconcile_procs
        ret_cache = os.path.join(self.opts['cachedir'], self.jid, _clean_tag(tag))
      File "/usr/local/lib/python3.6/site-packages/salt/state.py", line 154, in _clean_tag
        return salt.utils.files.safe_filename_leaf(tag)
      File "/usr/local/lib/python3.6/site-packages/salt/utils/files.py", line 313, in safe_filename_leaf
        return re.sub(u'[\\\\:/*?"<>|]', _replace, file_basename, flags=re.UNICODE)
      File "/usr/local/lib/python3.6/re.py", line 191, in sub
        return _compile(pattern, flags).sub(repl, string, count)
      File "/usr/local/lib/python3.6/site-packages/salt/utils/files.py", line 306, in _replace
        return urllib.quote(re_obj.group(0), safe=u'')
    AttributeError: module 'urllib' has no attribute 'quote'
```
### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
```FreeBSD11.1```
### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)
It only happens when I have ```- parallel: true``` in zabbix_host state.

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

```
Salt Version:
           Salt: 2017.7.4

Dependency Versions:
           cffi: 1.11.5
       cherrypy: unknown
       dateutil: 2.7.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: 1.3.12
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.5 (default, Apr  5 2018, 13:37:49)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.3

System Versions:
           dist:
         locale: UTF-8
        machine: amd64
        release: 11.1-RELEASE-p9
         system: FreeBSD
        version: Not Installed
```

**Comment by Ch3LL** [comment]
looks like this should be fixed on the had of 2017.7. There was this fix: https://github.com/saltstack/salt/pull/46023

particularly the change in teh file: salt/utils/files.py

Can you give that file change a try?

**Comment by deuscapturus** [comment]
I was able to resolve this issue by pulling over the changes to salt/utils/files.py from  #46023

**Comment by Ch3LL** [comment]
awesome thnaks for testing that. I will go ahead and close this and the fix will be in the next 2017.7.6 release :)
